targets/TARGET_STM/lp_ticker.c

```diff
@@ -194,21 +191,22 @@ void lp_ticker_set_interrupt(timestamp_t timestamp)
     LptimHandle.Instance = LPTIM1;
     irq_handler = (void (*)(void))lp_ticker_irq_handler;
 
+    __HAL_LPTIM_CLEAR_FLAG(&LptimHandle, LPTIM_FLAG_CMPOK);
+    __HAL_LPTIM_COMPARE_SET(&LptimHandle, timestamp);
     /* CMPOK is set by hardware to inform application that the APB bus write operation to the LPTIM_CMP register has been successfully completed */
     /* Any successive write before the CMPOK flag be set, will lead to unpredictable results */
     while (__HAL_LPTIM_GET_FLAG(&LptimHandle, LPTIM_FLAG_CMPOK) == RESET) {
```

Comment:
Will this bring back the blocking behavior?

Reply:
The change in #8129 caused blocking behavior as mentioned by @LMESTM.

Reply:
@c1728p9 my mistake - Jérôme explained to me and I actually misunderstood CMPOK (programing ok) with CMPM (comparator match) - looks ok to me now